From afb460d9fd6799f6362455bbe1fe476fabe401d6 Mon Sep 17 00:00:00 2001
Message-Id: <afb460d9fd6799f6362455bbe1fe476fabe401d6.1600853895.git.liu.denton@gmail.com>
In-Reply-To: <cover.1600853895.git.liu.denton@gmail.com>
References: <cover.1600427894.git.liu.denton@gmail.com>
	<cover.1600853895.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Thu, 17 Sep 2020 02:07:51 -0700
Subject: [PATCH v2 2/3] hooks--pre-push.sample: use hash-agnostic zero OID
To: Git Mailing List <git@vger.kernel.org>
Cc: brian m. carlson <sandals@crustytoothpaste.net>,
    Eric Sunshine <sunshine@sunshineco.com>,
    Taylor Blau <me@ttaylorr.com>,
    Junio C Hamano <gitster@pobox.com>,
    Chris Torek <chris.torek@gmail.com>,
    Andreas Schwab <schwab@linux-m68k.org>

The pre-push sample hook has the zero OID hardcoded as 40 zeros.
However, with the introduction of SHA-256 support, this assumption no
longer holds true. Replace the hardcoded $z40 with a call to

	git hash-object --stdin </dev/null | tr '[0-9a-f]' '0'

so the sample hook becomes hash-agnostic.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 templates/hooks--pre-push.sample | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/templates/hooks--pre-push.sample b/templates/hooks--pre-push.sample
index d0f30190ac..4ce688d32b 100755
--- a/templates/hooks--pre-push.sample
+++ b/templates/hooks--pre-push.sample
@@ -22,16 +22,16 @@
 remote="$1"
 url="$2"
 
-z40=0000000000000000000000000000000000000000
+zero=$(git hash-object --stdin </dev/null | tr '[0-9a-f]' '0')
 
 while read local_ref local_oid remote_ref remote_oid
 do
-	if test "$local_oid" = $z40
+	if test "$local_oid" = "$zero"
 	then
 		# Handle delete
 		:
 	else
-		if test "$remote_oid" = $z40
+		if test "$remote_oid" = "$zero"
 		then
 			# New branch, examine all commits
 			range="$local_oid"
-- 
2.28.0.760.g8d73e04208

